Neuro/Impostor/ImpostorHandler.cs

```diff
@@ -183,24 +178,22 @@ public IEnumerator Vent(Vent original)
             // use a random time between vent moves to make it more realistic
             yield return new WaitForSeconds(UnityEngine.Random.RandomRange(0.8f, 1.2f));
             bool playerFound = false;
-            foreach (KeyValuePair<PlayerControl, LastSeenPlayer> player in NeuroPlugin.Instance.Vision.GetPlayerLocations())
+            foreach ((PlayerControl player, LastSeenPlayer lastSeen) in NeuroPlugin.Instance.Vision.PlayerLocations)
             {
-                if (PlayerControl.LocalPlayer == player.Key) continue;
-                if (player.Key.Data.IsDead) continue;
-                if (player.Key.Data.RoleType == AmongUs.GameOptions.RoleTypes.Impostor) continue;
+                if (player.AmOwner) continue;
+                if (player.Data.IsDead) continue;
+                if (player.Data.Role.IsImpostor) continue;
 
                 // if we see a player in our radius, try a different vent
-                if (player.Value.time > Time.timeSinceLevelLoad - (2 * Time.fixedDeltaTime))
+                if (lastSeen.time > Time.timeSinceLevelLoad - (2 * Time.fixedDeltaTime))
                 {
-                    Info($"Spotted {player.Key.name}, trying a different exit vent...");
+                    Info($"Spotted {player.name}, trying a different exit vent...");
                     Vent next;
-                    while (true)
+                    do
                     {
                         possibleVents = GetAvailableNearbyVents(current);
                         next = possibleVents[UnityEngine.Random.RandomRangeInt(0, possibleVents.Count)];
-                        if (current == next) continue;
-                        break;
-                    }
+                    } while (current == next);
```

Comment:
```suggestion
                        possibleVents = GetAvailableNearbyVents(current);
                        next = possibleVents[UnityEngine.Random.RandomRangeInt(0, possibleVents.Count)];
```

`current` can never be equal to `next` because a vent will never have itself as its neighbour.

Reply:
Good to know. Thought I had encountered that issue in my testing before a while ago but if it doesn't then I'll remove it.

Reply:
Let me know if you encounter that issue again, it sounds very strange.